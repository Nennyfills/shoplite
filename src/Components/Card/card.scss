.card {
  text-align: center;
  margin: 40px auto;
  padding: 0.5em;
  background-color: #e3e6e3;
  color: #5d5f5d;
  width: 352px;
  .img {
    margin: auto;
    margin-bottom: 2em;
    object-fit: contain;
    img {
      width: 60%;
      height: 10em;
    }
  }
  .desc {
    text-transform: capitalize;
    h6 {
      margin-bottom: 0.4em;
    }
    h4 {
      margin-top: 0.4em;
    }
  }
  .card-btn {
    display: flex;
    justify-content: center;
  }
}
