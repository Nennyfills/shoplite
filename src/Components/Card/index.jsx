import React from 'react';
import PropTypes from 'prop-types';

import Button from 'Components/Button';

import './card.scss';

const Card = ({ src, title, currency, price, btnName, btnOnClick }) => {
  return (
    <>
      <div className="card" data-test="card">
        <div className="img">
          <img src={src} alt="Product" />
        </div>
        <div className="desc">
          <h6>{title}</h6>
          <h4>
            from {currency}
            <span>{price?.toFixed(2)}</span>
          </h4>
        </div>
        <div className="card-btn">
          <Button name={btnName} onClick={btnOnClick} />
        </div>
      </div>
    </>
  );
};

Card.defaultProps = {
  src: 'image',
  title: 'Cream',
  price: 0,
  currency: 'NGN',
};

Card.propTypes = {
  btnOnClick: PropTypes.func.isRequired,
  btnName: PropTypes.string.isRequired,
  src: PropTypes.string,
  title: PropTypes.string,
  price: PropTypes.number,
  currency: PropTypes.string,
};

export default Card;
