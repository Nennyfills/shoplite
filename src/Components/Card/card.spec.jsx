import React from 'react';
import { shallow } from 'enzyme';

import { checkProps} from 'Test/propsTypes';
import Card from '.';

const props = {
    btnOnClick: jest.fn(), 
    btnName: 'btnName',
    src: 'image',
    title: 'Cream',
    price: 0,
    currency: 'NGN',
}

const setUp = (props = {}) => {
    const component = shallow(<Card {...props} />);
    return component;
}

describe('Card', () => {
    let component;
    beforeEach(() => {
        component = setUp(props);
    })
    it('Should render without errors', () => {
        expect(component).toMatchSnapshot(); 
    });
    it('Should render card div', () => {
        const wrapper = component.find(`[data-test='card']`);
        expect(wrapper.length).toBe(1); 
    })
    it('Checking PropTypes should not throw error', () => {
        const propErr = checkProps(Card, props);
        expect(propErr).toBeUndefined();
    })
})
