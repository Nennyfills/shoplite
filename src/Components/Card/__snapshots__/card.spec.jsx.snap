// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should render without errors 1`] = `ShallowWrapper {}`;
